xbmcgui.py

```diff
@@ -262,11 +261,93 @@ def numeric(self, type, heading, defaultt=None, bHiddenInput=None):
+        0                         ShowAndGetDirectory
+        1                         ShowAndGetFile
+        2                         ShowAndGetImage
+        3                         ShowAndGetWriteableDirectory
+        ========================  =========================================
+
+        ===========================  ============================================================================
+        Shares                       Description
+        ===========================  ============================================================================
+        "programs"                   list program addons
+        "video"                      list video sources
+        "music"                      list music sources
+        "pictures"                   list picture sources
+        "files"                      list file sources (added through filemanager)
+        "games"                      list game sources
+        "local"                      list local drives
+        ""                           list local drives and network shares
+        ===========================  ============================================================================
+
+        :return: If enableMultiple is False (default): returns filename and/or path as a string to the location of the highlighted item, if user pressed 'Ok' or
+                 a masked item was selected. Returns the default value if dialog was canceled.
+                 If enableMultiple is True: returns tuple of marked filenames as a string if user pressed 'Ok' or a masked item was selected. Returns empty
+                 tuple if dialog was canceled.
+                 If type is 0 or 3 the enableMultiple parameter is ignored.
+
+        :rtype: str
+
+        """
+        if not self.is_interactive:
+            keyboard = self.get_keyboard_stub()
+            value = keyboard.get_next_input()
+            return value if value is not None else defaultt
+
+        KodiStub.print_heading(heading)
+        try:
+            answer = self.read_input("Please enter the path to a file [{}]?".format(defaultt), color=Colors.Yellow)
+            return answer if answer is not '' else defaultt
+        except EOFError:
+            return ''
+
+    def browseMultiple(self, type, heading, shares, mask='', useThumbs=False, treatAsFolder=False, defaultt=''):
+        """ Browser dialog for multiple files.
+
+        See xbmcgui.browse.
+
+        :return: Returns tuple of marked filenames as a string, if user pressed 'Ok' or a masked item was selected.
+                 Returns empty tuple if dialog was canceled.
+        :rtype: str
+
+        """
+        return self.browse(type, heading, shares, mask, useThumbs, treatAsFolder, defaultt, True)
+
+    def browseSingle(self, type, heading, shares, mask='', useThumbs=False, treatAsFolder=False, defaultt=''):
+        """ Browser dialog for single files.
+
```

Comment:
Could you please provide the correct parameter documentation in reStructuredText style? This helps debugging and type checking.